src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -2672,7 +2672,7 @@ private void CopyBatchesAsyncContinuedOnError(bool cleanupParser)
         }
 
         // Cleans the stateobj. Used in a number of places, specially in  exceptions.
-        private void CleanUpStateObjectOnError()
+        private void CleanUpStateObject(bool isCancelRequested)
```

Comment:
Can you make `isCancelRequested` optional with default value of `true`? This way all call sites wont have to pass in the parameter and only the impacted callsites pass in the `false` value to override default behavior?